Source/utils/utf8.hpp

```diff
@@ -29,14 +29,44 @@ inline char32_t ConsumeFirstUtf8CodePoint(string_view *input)
 	return result;
 }
 
+/**
+ * Returns true if this is a byte that potentially starts a valid UTF-8 sequence.
+ *
+ * Well-formed UTF-8 sequences are described in table 3-7 of S3.9 of the Unicode Standard
+ * see: https://www.unicode.org/versions/Unicode14.0.0/ch03.pdf#G7404
+ *
+ * This is not an inverse of IsTrailUtf8CodeUnit. Byte values C0-C1, F5-FF are not valid anywhere in a UTF-8 sequence.
+ */
+constexpr bool IsLeadUtf8CodeUnit(char x)
+{
+	// single byte character || multibyte character leader
+	return (x >= '\x00' && x <= '\x7F') || (x >= '\xC2' && x <= '\xF4');
+}

```

Comment:
Taking signed char explicitly here would also have triggered implementation defined behaviour. Thankfully clang and gcc both generate identical instructions for `signed char x; x >= 0 || ( x >= -62 && x <= -12);` and `char x; (x >= 0 && x <= 127) || (x >= -62 && x <= -12);`

MSVC generates different code but ends up with the same instruction count both ways, so figure it's close enough. Not gonna bother with performance testing this since it's not even used with the latest commit.